sdk/resources/azure-mgmt-resource/CHANGELOG.md

```diff
@@ -1,5 +1,24 @@
 # Release History
 
+## 24.0.0 (2025-06-16)
+
+### Features Added
+
+  - Model Deployment has a new parameter identity
+  - Model DeploymentParameter has a new parameter expression
+  - Model DeploymentProperties has a new parameter external_input_definitions
+  - Model DeploymentProperties has a new parameter external_inputs
+  - Model DeploymentWhatIfProperties has a new parameter external_input_definitions
+  - Model DeploymentWhatIfProperties has a new parameter external_inputs
+  - Model WhatIfChange has a new parameter extension
+
+### Breaking Changes
+
+  - The `deploymentstacks` module has been separated from the `azure-mgmt-resource` package into independent package `azure-mgmt-resource-deploymentstacks`
+  - The `deploymentscripts` module has been separated from the `azure-mgmt-resource` package into independent package `azure-mgmt-resource-deploymentscripts` 
+  - The `templatespecs` module has been separated from the `azure-mgmt-resource` package into independent package `azure-mgmt-resource-templatespecs`
+  - If you previously imported these modules using statements like `from azure.mgmt.resource import DeploymentStacksClient`, you'll need to update your code to `from azure.mgmt.resource.deploymentstacks import DeploymentStacksClient`. If you were already using the fully qualified import path, no changes are required. This applies to all three modules: `deploymentstacks`, `deploymentscripts`, and `templatespecs`.
+
```

Comment:
Revert https://github.com/Azure/azure-sdk-for-python/pull/41474/commits/e85b55dc3d2799881576545e567b9f458684fcc0 since it doesn't work: https://gist.github.com/lmazuel/e80aa8f9f282892b4e629a5ac0274744?permalink_comment_id=5610372#gistcomment-5610372.

Considering that we declare the change and migration in changelog, it shall not confuse SDK users too much.
@lmazuel for awareness.